rust_src/src/frames.rs

```diff
@@ -683,6 +685,39 @@ pub fn frame_face_alist(frame: LispFrameLiveOrSelected) -> LispObject {
     frame_ref.face_alist
 }
 
+/// Update the display_time slot of the buffers shown in WINDOW and all its descendants.
+fn make_frame_visible_1(window: LispWindowRef) {
+    if window.is_null() {
+        return;
+    }
+    for win in window.iter() {
+        let contents: LispObject = win.contents.into();
+        if contents.is_window() {
+            make_frame_visible_1(contents.into());
+        } else {
+            let mut buffer: LispBufferRef = contents.into();
+            buffer.set_display_time(current_time());
+        }
+    }
+}
+
+/// Make the frame FRAME visible (assuming it is an X window).
+/// If omitted, FRAME defaults to the currently selected frame.
+#[lisp_fn(min = "0")]
+pub fn make_frame_visible(frame: LispObject) {
+    #[cfg(feature = "window-system")]
+    {
+        let mut frame_ref: LispFrameRef = frame.into();
+        if frame_ref.is_gui_window() {
+            unsafe {
+                x_make_frame_visible(frame_ref.as_mut());
+            }
+        }
+    }
+    let frame_ref: LispFrameRef = frame.into();
```

Comment:
Maybe if we did this at the top? Then there would not be a reason to repeat it.

Reply:
The thing is that the `cfg` block needs a mutable pointer in the call to `x_make_window_visible`. However, when that `cfg` block is not compiled, the `mut` of `frame_ref` is unnecessary. I have a new commit to solve this issue without calling `into` multiple times.